src/main/scala/se/nimsa/sbx/storage/S3Storage.scala

```diff
@@ -45,18 +45,19 @@ class S3Storage(val bucket: String, val s3Prefix: String, val region: String)(im
   // Environment Variables - AWS_ACCESS_KEY_ID and AWS_SECRET_KEY
   // Java System Properties - aws.accessKeyId and aws.secretKey
   // Instance profile credentials delivered through the Amazon EC2 metadata service
-  val s3 = AmazonS3ClientBuilder.standard()
+  private val s3 = AmazonS3ClientBuilder.standard()
     .withRegion(region)
     .withCredentials(new DefaultAWSCredentialsProviderChain())
     .withClientConfiguration(new ClientConfiguration().withProtocol(Protocol.HTTPS))
     .build()
 
-  private val s3Stream = S3Client(credentialsFromProviderChain(), region)
-
   private def credentialsFromProviderChain() = {
     val providerChain = new DefaultAWSCredentialsProviderChain()
     val creds = providerChain.getCredentials
-    BasicCredentials(creds.getAWSAccessKeyId, creds.getAWSSecretKey)
+    creds match {
+      case sc: AWSSessionCredentials => akka.stream.alpakka.s3.auth.AWSSessionCredentials(sc.getAWSAccessKeyId, sc.getAWSSecretKey, sc.getSessionToken)
```

Comment:
Is it possible to simply return `sc` in this case or is there some reason the `AWSSessionCredentials` are recreated?